### DISCUSSION TITLE
Integrate this SDK in a C project

### DISCUSSION THREAD

**Discussion by disa6302** [aws/aws-sdk-cpp#2947] [Q&A] [answered]
Hello,

I am looking to integrate this SDK into a C application. I see 2 options: 

1. Create wrappers for the functions I use from this library to use in the C project
2. Write a C++ equivalent application

I already have the C++ application which works fine, but I do not want to maintain 2 copies because it causes parity maintenance burden.

Is there any other way I can achieve this? Has anybody tried to achieve this yet? 

Or if there is a C equivalent of this SDK, that would be awesome too! But I doubt it exists. Any pointers will be appreciated!

**Comment by jmklix** [comment, marked as answer]
Sorry, but there is no C sdk that is equivalent to this one. You would have to write C wrappers for the functions that you want to use or write an equivalent C++ application. 

My suggestion would be to write your application in C++ if possible. You can potentially use your current C code in a C++ application. It would be a lot of work to make wrappers with little benefit other than keeping your current C code.

**Reply by disa6302** [reply]
Got it. That is what I thought. Wanted to make sure I was not missing something

Thank you!
